module/os/linux/zfs/zfs_ioctl_os.c

```diff
@@ -269,12 +271,59 @@ openzfs_fini(void)
 	    ZFS_META_VERSION, ZFS_META_RELEASE, ZFS_DEBUG_STR);
 }
 
+
+extern int __init zcommon_init(void);
+extern void zcommon_fini(void);
+
+static int __init
+openzfs_init(void)
+{
+	int err;
+	if ((err = zcommon_init()) != 0)
+		goto zcommon_failed;
+	if ((err = icp_init()) != 0)
+		goto icp_failed;
+	if ((err = zstd_init()) != 0)
+		goto zstd_failed;
+	if ((err = openzfs_init_os()) != 0)
+		goto openzfs_os_failed;
```

Comment:
Sorry I'm late - just got to merge this into my repo. I thought it was nice of you guys to add a `openzfs_init_os()` so platforms can add their own... then I noticed `openzfs_init()` is inside `os/linux/zfs` and not shared sources. Shouldn't it have been in shared sources?

Reply:
My recollection is the `openzfs_init()` declaration ends up needing to be a little different between platforms which was a wrinkle.  But yes, if possible I think it'd be nice to move it to common sources.